code/parachain/frame/composable-support/src/types/rational.rs

```diff
@@ -0,0 +1,213 @@
+use core::num::NonZeroU64;
+
+use codec::{Decode, Encode, MaxEncodedLen};
+use frame_support::{
+	traits::{Get, TypedGet},
+	RuntimeDebug,
+};
+use scale_info::TypeInfo;
+
+use super::const_assertions::AssertNonZero;
+
+#[derive(
+	Copy,
+	Clone,
+	PartialEq,
+	Eq,
+	PartialOrd,
+	Ord,
+	RuntimeDebug,
+	Encode,
+	Decode,
+	MaxEncodedLen,
+	TypeInfo,
+)]
+/// Represents a [rational number], with 64 bit numbers for the numerator and denominator.
+///
+/// [rational number]: https://en.wikipedia.org/wiki/Rational_number
```

Comment:
Unsure why clippy doesn't pick this up, but it is more common to have the doc comment on top of the derives correct?

Reply:
Looks like this is still an in-progress feature: https://github.com/rust-lang/rustfmt/issues/3744

I will reorder it though :ok_hand:

Reply:
Fixed in https://github.com/ComposableFi/composable/commit/350c3a12e489d6e88eb850624b64c46bacdc4170